django/db/backends/base/features.py

```diff
@@ -302,6 +302,9 @@ class BaseDatabaseFeatures:
     # Does the backend support boolean expressions in SELECT and GROUP BY
     # clauses?
     supports_boolean_expr_in_select_clause = True
+    # Does the backend support comparing boolean expressions in WHERE clauses?
+    # Eg: WHERE (price > 0) IS NOT NULL
+    supports_boolean_expr_in_where_clause = True
```

Comment:
This name is not precise, Oracle supports `EXISTS(...)` or filters in the `WHERE` clause, see:

https://github.com/django/django/blob/1b08e9bf7d88dbad9324f462e5fc7ec582aef3cc/django/db/backends/oracle/operations.py#L713-L726

 ticket-33996 is related with a special handling of `NULL` values in check constraints that cannot be easily emulated on Oracle.

Maybe
```suggestion
    supports_boolean_expr_in_coalesce = True
```

Reply:
@felixxm Yep I know that boolean expressions "work" (for a given use case) in WHERE clauses. My thinking was not to be too specific as it's not just coalesce that doesn't work… ie you can't do this either:

```
WHERE (<boolean-expr>) IS NOT NULL
```

Knowing this did you still want to proceed with `supports_boolean_expr_in_coalesce`?  I mean the name is accurate but it's also quite specific … 🤷‍♂️☺️

Reply:
(also come to think of it my `supports_boolean_expr_in_where_clause` is misleading)

Reply:
I was about to suggest something like `supports_comparing_boolean_expr` too

> BOOLEAN data type is generally not supported in SQL on Oracle

Ha believe me I've become quite knowledgable on this topic as I spent most of the weekend looking into this 😅

> However it's possible when wrapped with CASE ... WHEN

_except_ for comparing with NULL yeah? 😁 (ie a condition evaluating to NULL is treated as false and the appropriate case's clause executed)

Reply:
@shangxiao Do you have time to keep working on this? If not, I can push final bits (release notes, docs warning for Oracle, and update the feature flag name).

Reply:
@felixxm Yep I'm just doing that now as we speak 👍